sql/core/src/test/scala/org/apache/spark/sql/InSetBenchmark.scala

```diff
@@ -0,0 +1,213 @@
+    val name = s"$numItems doubles"
+    val values = 1.0 to numItems by 1.0
+    val df = spark.range(1, numRows).select($"id".cast(DoubleType))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def smallDecimalBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems small decimals"
+    val values = (1 to numItems).map(v => s"CAST($v AS decimal(12, 1))")
+    val df = spark.range(1, numRows).select($"id".cast(DecimalType(12, 1)))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def largeDecimalBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems large decimals"
+    val values = (1 to numItems).map(v => s"9223372036854775812.10539$v")
+    val df = spark.range(1, numRows).select($"id".cast(DecimalType(30, 7)))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def stringBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems strings"
+    val values = (1 to numItems).map(n => s"'$n'")
+    val df = spark.range(1, numRows).select($"id".cast(StringType))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def timestampBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems timestamps"
+    val values = (1 to numItems).map(m => s"CAST('1970-01-01 01:00:$m' AS timestamp)")
+    val df = spark.range(1, numRows).select($"id".cast(TimestampType))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def dateBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems dates"
+    val values = (1 to numItems).map(n => 1970 + n).map(y => s"CAST('$y-01-01' AS date)")
+    val df = spark.range(1, numRows).select($"id".cast(TimestampType).cast(DateType))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def arrayBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems arrays"
+    val values = (1 to numItems).map(i => s"array($i)")
+    val df = spark.range(1, numRows).select(array($"id").as("id"))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def structBenchmark(numItems: Int, numRows: Long, minNumIters: Int): Benchmark = {
+    val name = s"$numItems structs"
+    val values = (1 to numItems).map(i => s"struct($i)")
+    val df = spark.range(1, numRows).select(struct($"id".as("col1")).as("id"))
+    benchmark(name, df, values, numRows, minNumIters)
+  }
+
+  def benchmark(
+      name: String,
+      df: DataFrame,
+      values: Seq[Any],
+      numRows: Long,
+      minNumIters: Int): Benchmark = {
+
+    val benchmark = new Benchmark(name, numRows, minNumIters, output = output)
+
+    df.createOrReplaceTempView("t")
+
+    def testClosure(): Unit = {
+      val df = spark.sql(s"SELECT * FROM t WHERE id IN (${values mkString ","})")
```

Comment:
`values mkString ","` -> `values.mkString(",")`

Reply:
Yep, will update that.